ui/src/app/components/rename-modal.tsx

```diff
@@ -54,6 +54,40 @@ export class RenameModal extends React.Component<Props, States> {
       saving: false,
     };
   }
+  generateNewNameForValidation(
```

Comment:
This is only used for validation, so we're now validating lower case files. I tested dataset rename in addition to notebook rename and I don't see any issues